pkg/migrations/migrations_test.go

```diff
@@ -502,3 +505,217 @@ func TestCreateSystemTable(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestUpdateViewDependenciesMigration(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	ctx := context.Background()
+
+	// Start up a test server without running the view update migration. We also
+	// hijack the migration process to capture the SQL memory metric object, needed below.
+	var memMetrics *sql.MemoryMetrics
+	backwardCompatibleMigrations = []migrationDescriptor{
+		{
+			name:   "create system.settings table",
+			workFn: createSettingsTable,
+		},
+		{
+			name: "capture mem metrics",
+			workFn: func(ctx context.Context, r runner) error {
+				memMetrics = r.memMetrics
+				return nil
+			},
+		},
+	}
+	var s serverutils.TestServerInterface
+	var kvDB *client.DB
+
+	t.Run("starting server", func(t *testing.T) {
+		s, _, kvDB = serverutils.StartServer(t, base.TestServerArgs{})
+	})
+
+	defer s.Stopper().Stop(ctx)
+	e := s.Executor().(*sql.Executor)
+
+	t.Run("create test tables", func(t *testing.T) {
```

Comment:
It's not clear to me that these subtests are providing any value, since they can't be run independently of one another. Plus, each subtest creates its own session, which adds quite a bit of boilerplate. @knz, does this exercise some codepath that wouldn't be executed if you used the same session throughout?